modules/account-lib/src/mpc/curves/secp256k1.ts

```diff
@@ -0,0 +1,51 @@
+import { ec } from 'elliptic';
+import BaseCurve from './baseCurve';
+
+const EC = new ec('secp256k1');
+const order = BigInt(EC.curve.n.toString());
+
+export class secp256k1 implements BaseCurve {
+  scalarRandom(): bigint {
+    return BigInt(0); // TODO
+  }
+
+  scalarReduce(s: bigint): bigint {
+    return s % order;
+  }
+
+  scalarNegate(s: bigint): bigint {
+    return -s % order;
+  }
+
+  scalarInvert(s: bigint): bigint {
+    return BigInt(0); // TODO
+  }
+
+  scalarAdd(x: bigint, y: bigint): bigint {
+    return (x + y) % order;
+  }
+
+  scalarSub(x: bigint, y: bigint): bigint {
```

Comment:
modulus operation for negative numbers in python strangely behaves differently from what Js does, So need to recheck this.

```
-5 % 15 ==> 10 // python
-5 % 15 ==>-5 // js
```

Reply:
The result of a mod operation should always produce a positive number. You don't want your private key to be a negative number. If the result `(x - y)` is less than zero, return `(x - y) + order`. Please also ensure that `scalarNegate` only returns positive numbers.